lms/djangoapps/grades/api/views.py

```diff
@@ -47,14 +47,21 @@ def _get_course(self, course_key_string, user, access_action):
                 user,
                 access_action,
                 course_key,
-                check_if_enrolled=True
+                check_if_enrolled=True,
             )
         except Http404:
             log.info('Course with ID "%s" not found', course_key_string)
             return self.make_error_response(
                 status_code=status.HTTP_404_NOT_FOUND,
                 developer_message='The user, the course or both do not exist.',
-                error_code='user_or_course_does_not_exist'
+                error_code='user_or_course_does_not_exist',
+            )
+        except CourseAccessRedirect:
+            log.info('User %s does not have access to course with ID "%s"', user.username, course_key_string)
+            return self.make_error_response(
+                status_code=status.HTTP_404_NOT_FOUND,
+                developer_message='User does not have access to course.',
+                error_code='course_access_denied',
```

Comment:
Who can see the developer_message and error_code?  Does this have our API expose authorization information that we are trying to hide.  Does the message and code need to match a 404 message (like above) to not give something away?